### PR TITLE
Mukuna solution 1

### PR DESCRIPTION
**Pull request by kerthnorth**
---
name: Password generator 
about: This PR contains my solution and test cases
---

### Files

- [ ] The file name describes the function's behavior
- [ ] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [ ] There is a module docstring in the tests file

### Unit Tests

- [ ] The test class has a helpful name in PascalCase
- [ ] The test class has a docstring
- Every unit test has
  - [ ] A helpful name
  - [ ] A clear docstring
  - [ ] Only one assertion
  - [ ] There is no logic in the unit test
- [ ] All tests pass
- [ ] There are tests for defensive assertions
- [ ] There are tests for boundary cases

### Function Docstring

- [ ] The function's behavior is described
- The function's arguments are described:
  - [ ] Type
  - [ ] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [ ] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [ ] Include 3 or more (passing!) doctests

### The Function

- [ ] The function's name describes it's behavior
- [ ] The function's name matches the file name
- [ ] The function has correct type annotations

### Do's

- [ ] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [ ] The function's strategy _is not_ described in the documentation
- [ ] Comments explain the _strategy_, **not** the _implementation_
- [ ] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [ ] The code passes the formatting checks
- [ ] The code passes all Ruff linting checks
- [ ] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [ ] The code follows the strategy as simply as possible
- [ ] The implementation is as simple as possible given the strategy
- [ ] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

**Review by MohammadRAlSalloum — DISMISSED** [review]
All of the cheklists points have been checked!
Great Work!

**Comment by MohammadRAlSalloum** [comment]
Kindly, Wait at least one more review before merging.

File: .vscode/settings.json
Comment (RaedAlyan) [inline .vscode/settings.json:1]: @kerthnorth this file shouldn't be pushed to the repo, please remove it

File: solutions/__init__.py
Comment (RaedAlyan) [inline solutions/__init__.py:1]: the same here, you don't have to push __init__.py because I've already pushed it before

File: solutions/password_generator.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Create a secure password with a mix of letters and numbers.
+
+    This function generates a random password using a combination of
+    letters and numbers. You firstly specify
+    the desired password lengths, however it must be between 4 and 10
+    characters long. If you provide an invalid input it will raise a value error.
+
+    Example:
+        >>> password_generator(8)
+        "MuKunas7"
```
Comment (RaedAlyan) [inline solutions/password_generator.py:23]: how did you know that the generated password would be MuKunas7? because you said that the password will be generated randomly, right?

File: solutions/tests/__init__.py
Comment (RaedAlyan) [inline solutions/tests/__init__.py:1]: please don't push this file

File: solutions/password_generator.py (hunk 1 above)
Comment (kerthnorth) [reply solutions/password_generator.py:23]: That is an example of the expected output if you request a password of length 8.

File: solutions/__init__.py
Comment (kerthnorth) [reply solutions/__init__.py:1]: If i remove it now some checks won't pass. But knowing that the code already functions we can remove it and still merge the code

Comment (RaedAlyan) [reply solutions/__init__.py:1]: because you made your PR before I merged my PR which contains the __init__.py file, right?

Comment (kerthnorth) [reply solutions/__init__.py:1]: Yes, but I have fixed it now. I deleted them and pulled from the main. Thank you.

**Review by MahmoudAlnouri — APPROVED** [review]
Looks good

**Review by RaedAlyan — APPROVED** [review]
This PR looks fine to me!
Good job @kerthnorth
